client/client.go

```diff
@@ -87,28 +106,47 @@ func New(config *Config) *Client {
 		config = &Config{}
 	}
 
-	// By default talk over a UNIX socket.
+	var client *Client
+	var transport *http.Transport
+
 	if config.BaseURL == "" {
-		transport := &http.Transport{Dial: unixDialer(config.Socket), DisableKeepAlives: config.DisableKeepAlive}
-		return &Client{
-			baseURL: url.URL{
-				Scheme: "http",
-				Host:   "localhost",
-			},
-			doer:      &http.Client{Transport: transport},
-			userAgent: config.UserAgent,
+		// By default talk over a UNIX socket.
+		transport = &http.Transport{Dial: unixDialer(config.Socket), DisableKeepAlives: config.DisableKeepAlive}
+		baseURL := url.URL{Scheme: "http", Host: "localhost"}
+		client = &Client{baseURL: baseURL}
+	} else {
+		// Otherwise talk regular HTTP-over-TCP.
+		baseURL, err := url.Parse(config.BaseURL)
+		if err != nil {
+			panic(fmt.Sprintf("cannot parse server base URL: %q (%v)", config.BaseURL, err))
```

Comment:
We should just return an error vs panicing.

Reply:
This panic was there already, but I agree -- fixed.